e2e/vaultstatefulset/vaultstatefulset_test.go

```diff
@@ -0,0 +1,296 @@
+	// unsealing configuration of the Vault deployment.
+	ct.PatchManifest(t, func(m manifest.Manifest) manifest.Manifest {
+		for key, policy := range m.Policies {
+			// TODO only do this for Vault image, not the other deployment
+			policy.WorkloadSecretID = "vault_unsealing"
+			m.Policies[key] = policy
+		}
+		return m
+	})
+
+	require.True(t, t.Run("apply", ct.Apply), "Kubernetes resources need to be applied for subsequent tests")
+
+	require.True(t, t.Run("set", ct.Set), "contrast set needs to succeed for subsequent tests")
+
+	require.True(t, t.Run("contrast verify", ct.Verify), "contrast verify needs to succeed for subsequent tests")
+
+	// Get Vault pod for subsequent tests.
+	ctx, cancel := context.WithTimeout(t.Context(), ct.FactorPlatformTimeout(2*60*time.Second))
+	defer cancel()
+	require.NoError(t, ct.Kubeclient.WaitFor(ctx, kubeclient.Ready, kubeclient.StatefulSet{}, ct.Namespace, "vault"))
+	pods, err := ct.Kubeclient.PodsFromOwner(ctx, ct.Namespace, "StatefulSet", "vault")
+	require.NoError(t, err)
+	require.Len(t, pods, 1)
+
+	var token string
+	loggingPath := "/vault/data/openbao.log"
+
+	t.Run("transit engine api auto-unseals Vault", func(t *testing.T) {
+		require := require.New(t)
+		ctx, cancel := context.WithTimeout(t.Context(), ct.FactorPlatformTimeout(2*60*time.Second))
+		defer cancel()
+
+		vaultLogs, stdErr, err := ct.Kubeclient.ExecContainer(ctx, ct.Namespace, pods[0].Name, "openbao-server", []string{"cat", loggingPath})
+		require.NoError(err, "stdout: %s, stderr: %s", vaultLogs, stdErr)
+
+		coordinatorLogs, err := ct.Kubeclient.GetContainerLogs(ctx, ct.Namespace, "StatefulSet", "coordinator", "coordinator")
+		require.NoError(err)
+
+		// Verifies that Vault is automatically unsealed using the coordinator's transit engine API.
+		require.True(checkUnsealingLogs(coordinatorLogs, vaultLogs, "vault_unsealing"))
+	})
+
+	// Extracts the root token from Vault logs, enables the KV secrets engine, and writes
+	// a test secret to verify that Vault accepts and stores data correctly.
+	// This secret will later be used to validate data persistence after a Vault restart.
+	t.Run("Enable KV engine and create secret on Vault", func(t *testing.T) {
+		require := require.New(t)
+		ctx, cancel := context.WithTimeout(t.Context(), ct.FactorPlatformTimeout(60*time.Second))
+		defer cancel()
+
+		baoClientLogs, err := ct.Kubeclient.GetContainerLogs(ctx, ct.Namespace, "StatefulSet", "vault", "openbao-client")
+		require.NoError(err)
+		token, err = extractVaultRootToken(baoClientLogs)
+		require.NoError(err, "failed to extract Vault root token from logs:\n%s", baoClientLogs)
+		require.NotEmpty(token, "extracted token is empty")
+
+		stdOut, stdErr, err := ct.Kubeclient.ExecContainer(
+			ctx,
+			ct.Namespace,
+			pods[0].Name,
+			"openbao-client",
+			[]string{
+				"sh",
+				"-c",
+				fmt.Sprintf("export VAULT_TOKEN=%s && bao secrets enable -path=mykv kv", token),
```

Comment:
Nit: you can simplify these scripts like this:

```suggestion
				fmt.Sprintf("VAULT_TOKEN=%s bao secrets enable -path=mykv kv", token),
```

Reply:
Considering that you're executing such scripts 4 times, defining a function that just takes the `bao` command might make sense.